### DISCUSSION TITLE
Disable authentication option

### DISCUSSION THREAD

**Discussion by gk-fschubert** [Azure/Azurite#2312] [General]
hi, 

sorry if that has been asked previously but i can't find a answer. 

Is there a way of just disabling the authentication? 
We need to use Azurite as storage for an other project in our unit tests. Therefore it would be really helpful to be not required to use libraries to generate the SAS token.

**Comment by blueww** [comment]
@gk-fschubert 

Azurite is Azure storage server emulator. As product Azure storage server not support disable authentication, we don't support it iether.
If you don't want to generate SAS token, how about using shared key authentication?

**Reply by gk-fschubert** [reply]
@blueww 

If I understand it correctly, even with shared key authentication it's not possible to have just one curl request where you not have to calculate something before. Or do i misunderstood something? 

Yes understand that but for a development where it's not intended to have the code ever reach out to the real storage endpoint, it would be really good to either be able to disable the authentication or to have a static API key which I can use in a curl request.

**Comment by blueww** [comment]
@gk-fschubert 

Thanks for the responds!
Yes, for shared key credential, the request need a signature signed by shared key.

As Azurite is open source, would you like to clone Azurite code, disable SAS validation, and build private Azurite for yourself (see https://github.com/Azure/Azurite#github)? 
You can disable blob service SAS validation by make the following 2 functions always return true: 
- Account sas: https://github.com/Azure/Azurite/blob/105a9cd217b881922016edb7a683ebea1ab3b28a/src/blob/authentication/AccountSASAuthenticator.ts#L26
- Blob service SAS: https://github.com/Azure/Azurite/blob/105a9cd217b881922016edb7a683ebea1ab3b28a/src/blob/authentication/BlobSASAuthenticator.ts#L33

**Reply by gk-fschubert** [reply]
Yes that's fine. 

Thanks :)
